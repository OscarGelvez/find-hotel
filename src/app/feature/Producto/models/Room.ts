export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
