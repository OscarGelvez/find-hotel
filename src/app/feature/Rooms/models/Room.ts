export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
