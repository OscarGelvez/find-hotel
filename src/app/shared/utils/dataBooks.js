export const bookInfo = {
  bookData: {
    name: 'oscar andres',
    email: 'oscargelvez23@gmail.com',
    identification_type: '1',
    identification: '1323131232',
  },
  roomData: {
    id: 54,
    title: 'Peace and Love',
    image:
      'https://dynamic-media-cdn.tripadvisor.com/media/photo-o/0d/2d/70/6d/habitacion-master-suite.jpg?w=600&h=300&s=1',
    hotel: {
      id: 3,
      name: 'Cencor hotel',
      address: 'Calle 1 # 16-23',
      stars: 4,
      description: 'Ambiente familiar, amigos, descaso y relajaci칩n',
      score: 7.2,
    },
    value: '56.000',
    ac: 'N',
    parking: 'N',
    available_from: '2021-05-02',
    available_until: '2021-05-04',
    wifi: 'N',
    capacity: 1,
    description: 'Ba침o privado, cama sencilla, TV.',
  },
  id: 5,
};

export const bookInfo2 = {
  bookData: {
    name: 'Oscar Gelvez Test',
    email: 'test@cypress.com',
    identification_type: '1',
    identification: '123456789',
  },
  roomData: {
    id: 50,
    title: 'Suite full. Caroline',
    image: 'https://static.abc.es/Media/201504/27/hotel12--644x362.jpg',
    hotel: {
      id: 1,
      name: 'Gold Hotel 5',
      address: 'Calle 54 # 56 -32 Av Libre',
      stars: 4,
      description:
        'Hotel con ambiente familiar, increible opci칩n. Muchas comodidades',
      score: 8.7,
    },
    value: '154.000',
    ac: 'Y',
    parking: 'Y',
    available_from: '2021-04-13',
    available_until: '2021-04-30',
    wifi: 'Y',
    capacity: 1,
    description: 'Ba침o privado, cama doble, con nevera, Wifi, TV.',
  },
  id: 1,
};
