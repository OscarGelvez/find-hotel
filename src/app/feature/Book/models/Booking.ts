import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
