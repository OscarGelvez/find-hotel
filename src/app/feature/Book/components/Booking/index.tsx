import { fieldsFormFindBooking } from 'app/feature/Rooms/models/Room';
import { Button } from 'app/shared/components/Button';
import { Field, Form, Formik, FormikHelpers } from 'formik';
import * as PropTypes from 'prop-types';
import * as React from 'react';
import * as Yup from 'yup';

interface BookingDetail {
  onFindBooking: (findBookingData: fieldsFormFindBooking) => void;
}

const initialValues = {
  email: '',
  identification_type: '',
  identification: '',
};

const validationSchema = Yup.object().shape<fieldsFormFindBooking>({
  email: Yup.string().email('Email no válido').required('Necesitamos tu email'),
  identification_type: Yup.string().required(
    'Necesitamos tu tipo de identificación'
  ),
  identification: Yup.string().required('Necesitamos tu identificación'),
});

export const BookingDetail: React.FC<BookingDetail> = ({ onFindBooking }) => {
  const handleSubmit = (
    values: fieldsFormFindBooking,
    { resetForm }: FormikHelpers<fieldsFormFindBooking>
  ) => {
    onFindBooking({
      email: values.email,
      identification_type: values.identification_type,
      identification: values.identification,
    });
    resetForm();
  };
  return (
    <>
      <div data-testid="find-rooms" className="card-rooms">
        <div className="row">
          <div className="card filter-area ">
            <div className="col">
              <Formik
                initialValues={initialValues}
                validationSchema={validationSchema}
                onSubmit={handleSubmit}
              >
                {({ errors, touched }) => (
                  <Form data-testid="form-find-rooms">
                    <div className="form-group row mb-3 mt-3">
                      <div className=" col-12 col-md-4">
                        <label
                          htmlFor="email"
                          className="col-12 col-form-label"
                        >
                          Correo electrónico
                        </label>
                        <Field
                          name="email"
                          type="email"
                          className="form-control col-12"
                          placeholder="eje: joe@email.com"
                        />
                        {errors.email && touched.email ? (
                          <small className="text-danger">{errors.email}</small>
                        ) : null}
                      </div>
                      <div className="col-12 col-md-4">
                        <label className="pr-2" htmlFor="identification_type">
                          Tipo de identificación
                        </label>
                        <Field
                          as="select"
                          className=" d-block w-100"
                          aria-label=".form-select-sm example"
                          id="identification_type"
                          name="identification_type"
                        >
                          <option value="">Seleccione</option>
                          <option value="1">Cédula de ciudadanía</option>
                          <option value="2">Cédula de extranjeria</option>
                          <option value="3">Nit</option>
                          <option value="4">Pasaporte</option>
                        </Field>
                        {errors.identification_type &&
                        touched.identification_type ? (
                          <small className="text-danger">
                            {errors.identification_type}
                          </small>
                        ) : null}
                      </div>
                      <div className=" col-12 col-md-4">
                        <label className="pr-2" htmlFor="identification">
                          Identificación
                        </label>
                        <Field
                          name="identification"
                          className="form-control"
                          placeholder="eje: 60830147"
                        />
                        {errors.identification && touched.identification ? (
                          <small className="text-danger">
                            {errors.identification}
                          </small>
                        ) : null}
                      </div>
                    </div>
                    <div className="row text-center">
                      <Button type="submit" className="btn ">
                        Buscar
                      </Button>
                    </div>
                  </Form>
                )}
              </Formik>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

BookingDetail.propTypes = {
  // msg: PropTypes.string.isRequired,
};
