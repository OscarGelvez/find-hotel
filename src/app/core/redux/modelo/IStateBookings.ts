import { IBooking } from '../../../feature/Book/models/Booking';

export interface IStateBookings {
  listBookings: IBooking[];
  deleteBookingId: number;
}
