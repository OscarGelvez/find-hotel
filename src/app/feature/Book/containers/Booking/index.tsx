import {
  IBooking,
  IFieldsFormFindBooking,
} from 'app/feature/Book/models/Booking';
import * as PropTypes from 'prop-types';
import * as React from 'react';

import { FindBookingForm } from '../../components/findBooking';
import { ListBookings } from '../../components/listBooking';
import { DivContainer } from './styles';

interface IBookingsProps {
  findBooking: (findBookingData: IFieldsFormFindBooking) => void;
  listBookingsFinded: Array<IBooking>;
  cancelBooking: (bookId: number) => void;
  isLoading: boolean;
}

export const Booking: React.FC<IBookingsProps> = ({
  findBooking,
  listBookingsFinded,
  cancelBooking,
  isLoading,
}) => {
  return (
    <DivContainer data-testid="book">
      <FindBookingForm onFindBooking={findBooking} />
      <ListBookings
        listBookingsFinded={listBookingsFinded}
        cancelBooking={cancelBooking}
        isLoading={isLoading}
      />
    </DivContainer>
  );
};

Booking.propTypes = {
  findBooking: PropTypes.func.isRequired,
  listBookingsFinded: PropTypes.array.isRequired,
  cancelBooking: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
};
