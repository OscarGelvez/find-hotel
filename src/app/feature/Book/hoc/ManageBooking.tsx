import { onFindBooking } from 'app/core/redux/actions/rooms/ActionsRooms';
import { Istate } from 'app/core/redux/modelo/GeneralState';
import { connect } from 'react-redux';

import { Booking } from '../containers/Booking';

const mapStateToProps = (state: Istate) => {
  return {
    listRooms: state.rooms.listRooms,
    selectedRoom: state.rooms.selectedRoomId,
  };
};

export const ManageBooking = connect(mapStateToProps, { onFindBooking })(
  Booking
);
