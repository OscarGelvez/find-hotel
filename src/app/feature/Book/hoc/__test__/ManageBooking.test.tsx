import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { IStateBookings } from '../../../../core/redux/modelo/IStateBookings';
import { bookInfo } from '../../../../shared/utils/dataBooks';
import { IBooking } from '../../models/Booking';
import { ManageBooking } from '../ManageBooking';

const mockStore = configureStore([]);

const newBook: IBooking = bookInfo;
const listBooks: IBooking[] = [newBook];

const bookings: IStateBookings = {
  listBookings: listBooks,
  deleteBookingId: -1,
};

describe('Prueba componente de orden superior HOC con redux', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      bookings: bookings,
    });

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ManageBooking />
        </BrowserRouter>
      </Provider>
    );
  });

  it('Compara snapshot del componente ManageRooms renderizado', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
