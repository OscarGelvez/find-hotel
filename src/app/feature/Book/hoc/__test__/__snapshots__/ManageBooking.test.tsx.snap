// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente de orden superior HOC con redux Compara snapshot del componente ManageRooms renderizado 1`] = `
<div
  className="sc-AxiKw bvTGbm"
  data-testid="book"
>
  <div
    className="card-rooms"
    data-testid="find-bookings"
  >
    <div
      className="row"
    >
      <div
        className="card filter-area "
      >
        <h4
          className="mt-3"
        >
          Encuentra tu reserva
        </h4>
        <div
          className="col"
        >
          <form
            action="#"
            data-testid="form-find-bookings"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="form-group row mb-3 mt-3"
            >
              <div
                className=" col-12 col-md-4"
              >
                <label
                  className="col-12 col-form-label"
                  htmlFor="email"
                >
                  Correo electrónico*
                </label>
                <input
                  className="form-control col-12"
                  data-testid="form-bookings-email"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="eje: joe@email.com"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="col-12 col-md-4"
              >
                <label
                  className="col-12 col-form-label"
                  htmlFor="identification_type"
                >
                  Tipo de identificación*
                </label>
                <select
                  aria-label=".form-select"
                  className=" d-block w-100"
                  data-testid="form-bookings-identification_type"
                  id="identification_type"
                  name="identification_type"
                  onBlur={[Function]}
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    Seleccione
                  </option>
                  <option
                    value="1"
                  >
                    Cédula de ciudadanía
                  </option>
                  <option
                    value="2"
                  >
                    Cédula de extranjeria
                  </option>
                  <option
                    value="3"
                  >
                    Nit
                  </option>
                  <option
                    value="4"
                  >
                    Pasaporte
                  </option>
                </select>
              </div>
              <div
                className=" col-12 col-md-4"
              >
                <label
                  className="col-12 col-form-label"
                  htmlFor="identification"
                >
                  Identificación*
                </label>
                <input
                  className="form-control"
                  data-testid="form-bookings-identification"
                  id="identification"
                  name="identification"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="eje: 60830147"
                  value=""
                />
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="mt-4 text-center"
              >
                <button
                  className="sc-AxjAm cYJbNq btn w-75"
                  data-testid="form-bookings-btn-submit"
                  type="submit"
                >
                  Buscar
                </button>
              </div>
              <div
                className="text-center mb-2 mt-1"
              >
                <button
                  className="sc-AxjAm cYJbNq button-link"
                  data-testid="form-bookings-btn-reset"
                  type="reset"
                >
                  limpiar
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-rooms"
    data-testid="list-bookings"
  >
    <div
      className="row pt-5"
    >
      <div
        className="col"
      >
        <div
          className="card room-item"
          data-testid="card-information-bookigns"
        >
          <div
            className="card-header book-item-header "
          >
            <h3>
              Cencor hotel
            </h3>
          </div>
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <div
                className="col-12 col-md-6 col-lg-3"
              >
                <h5>
                  A nombre de:
                </h5>
                <span>
                  oscar andres
                </span>
                <h5
                  className="mt-2"
                >
                  Ingreso:
                </h5>
                <span>
                  2021-05-02
                </span>
              </div>
              <div
                className="col-12 col-md-6 col-lg-3"
              >
                <h5>
                  Habitación:
                </h5>
                <span>
                  Peace and Love
                </span>
                <h5
                  className="mt-2"
                >
                  Salida:
                </h5>
                <span>
                  2021-05-04
                </span>
              </div>
              <div
                className="col-12 col-md-6 col-lg-3"
              >
                <h5>
                  Valor:
                </h5>
                <span>
                  56.000
                </span>
                <h5
                  className="mt-2"
                >
                  Camas:
                </h5>
                <span>
                  1
                </span>
              </div>
              <div
                className="col-12 col-md-6 col-lg-3"
              >
                <div
                  aria-labelledby="modalDetailBookLabel"
                  className="modal fade"
                  data-testid="modal-detail-book"
                  id="modalDetailBook"
                >
                  <div
                    className="modal-dialog modal-xl"
                  >
                    <div
                      className="modal-content"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                          id="modalDetailBookLabel"
                        >
                          Detalles de tu reserva
                        </h5>
                        <button
                          aria-label="Close"
                          className="btn-close"
                          data-bs-dismiss="modal"
                          type="button"
                        />
                      </div>
                      <div
                        className="modal-body"
                      />
                    </div>
                  </div>
                </div>
                <button
                  className="btn btn-primary btn-book w-100"
                  data-bs-target="#modalDetailBook"
                  data-bs-toggle="modal"
                  data-testid="list-book-btn-detail"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="fas fa-check ml-3"
                  />
                   Detalles
                </button>
                <button
                  className="btn btn-danger btn-book w-100 mt-3"
                  data-bs-target="#modalDeleteBook"
                  data-bs-toggle="modal"
                  data-testid="list-book-btn-cancel"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="fas fa-times ml-3"
                  />
                   Cancelar
                </button>
                <div
                  aria-labelledby="modalDeleteBookLabel"
                  className="modal fade"
                  data-testid="modal-book-delete"
                  id="modalDeleteBook"
                >
                  <div
                    className="modal-dialog modal-xl"
                  >
                    <div
                      className="modal-content"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                          id="modalDeleteBookLabel"
                        >
                          Eliminar reserva
                        </h5>
                        <button
                          aria-label="Close"
                          className="btn-close"
                          data-bs-dismiss="modal"
                          type="button"
                        />
                      </div>
                      <div
                        className="modal-body"
                      />
                      <div
                        className="modal-footer"
                      >
                        <div
                          className="d-flex justify-content-between w-100"
                        >
                          <button
                            className="btn btn-secondary w-25"
                            data-bs-dismiss="modal"
                            data-testid="modal-book-delete-no"
                            type="button"
                          >
                            No
                          </button>
                          <button
                            className="btn btn-danger w-25"
                            data-testid="modal-book-delete-cancel"
                            onClick={[Function]}
                            type="button"
                          >
                            Sí, cancelar
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
